app/theme-initializer.tsx

```
@@ -0,0 +1,91 @@
+        if (themeParam) {
+          localStorage.setItem('theme-preference', themeParam);
+        }
+      } catch (e) {
+        console.error('Error in theme initialization script:', e);
+      }
+    })();
+  `;
+
+  return (
+    <script
+      dangerouslySetInnerHTML={{ __html: themeScript }}
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)